src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+    private static final String INSERT_BOOK = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String SELECT_BOOK_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String SELECT_ALL_BOOKS = "SELECT * FROM books";
+    private static final String UPDATE_BOOK = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BOOK_BY_ID = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        String sql = INSERT_BOOK;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                            sql, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            checkIfExecuted(statement);
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create a book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = SELECT_BOOK_BY_ID;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        String sql = SELECT_ALL_BOOKS;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
```

Comment:
Similar to the previous comment, consider refactoring the code that retrieves data from `ResultSet` into a separate private method to avoid code duplication. This will improve code maintainability and readability. This suggestion aligns with the checklist item: 'Try to avoid code duplication. Especially, when you are working with ResultSet. Move retrieving data from ResultSet into Entity object to a separate private method.'